src/cdomain/value/cdomains/intDomain.ml

```diff
@@ -1382,8 +1393,9 @@ struct
       let min_ys = minimal ys |> Option.get in
       let max_ys = maximal ys |> Option.get in
       let min_range,max_range = range ik in
-      let min = if min_xs =. min_range then min_ys else min_xs in
-      let max = if max_xs =. max_range then max_ys else max_xs in
+      let threshold = get_interval_threshold_widening () in
+      let min = if min_xs =. min_range || threshold && min_ys <. min_xs && IArith.is_lower_threshold min_xs then min_ys else min_xs in
```

Comment:
I think they are flipped here
```suggestion
      let min = if min_xs =. min_range || threshold && min_xs <. min_ys && IArith.is_lower_threshold min_xs then min_ys else min_xs in
```

Reply:
You're absolutely right. I've added a test for interval set that would have caught this mistake.